tests/integration/utils/launch_tests_for_content.js

```diff
@@ -202,104 +197,123 @@ export default function launchTestsForContent(
           const allAdaptationInfos = periodInfos.adaptations;
           const allAdaptations = period.adaptations;
 
+          const promisesToResolve = [];
           Object.keys(allAdaptations).forEach((type) => {
-            const adaptations = allAdaptations[type];
-            const adaptationsInfos = allAdaptationInfos[type];
-
-            expect(adaptations.length).to.equal(adaptationsInfos.length);
-
-            for (
-              let adaptationIndex = 0;
-              adaptationIndex < adaptations.length;
-              adaptationIndex++
-            ) {
-              const adaptation = adaptations[adaptationIndex];
-              const adaptationInfos = adaptationsInfos[adaptationIndex];
-              const bitrates = adaptationInfos.representations
-                .map(representation => representation.bitrate);
-
-              expect(!!adaptation.isAudioDescription)
-                .to.equal(!!adaptationInfos.isAudioDescription);
-
-              expect(!!adaptation.isClosedCaption)
-                .to.equal(!!adaptationInfos.isClosedCaption);
-
-              expect(adaptation.language)
-                .to.equal(adaptationInfos.language);
-
-              expect(adaptation.normalizedLanguage)
-                .to.equal(adaptationInfos.normalizedLanguage);
-
-              expect(adaptation.getAvailableBitrates())
-                .to.eql(bitrates);
-
-              expect(typeof adaptation.id).to.equal("string");
-              expect(adaptation.representations.length)
-                .to.equal(adaptationInfos.representations.length);
-
-              for (
-                let representationIndex = 0;
-                representationIndex < adaptation.representations.length;
-                representationIndex++
-              ) {
-                const representation = adaptation
-                  .representations[representationIndex];
-                const representationInfos = adaptationInfos
-                  .representations[representationIndex];
+            promisesToResolve.push(
```

Comment:
1. what are you doing?
2. Why not just using `.map` without he need to create a `promisesToResolve` variable?